csrc/moe/marlin_moe_wna16/generate_kernels.py

```diff
@@ -0,0 +1,87 @@
+# SPDX-License-Identifier: Apache-2.0
+import itertools
+import os
+
+import jinja2
+
+FILE_HEAD = """
+// auto generated by generate.py
+// clang-format off
+
+#include "kernel.h"
+#include "marlin_template.h"
+
+namespace MARLIN_NAMESPACE_NAME {
+""".strip()
+
+TEMPLATE = ("template __global__ void Marlin<"
+            "{{scalar_t}}, "
+            "{{w_type_id}}, "
+            "{{threads}}, "
+            "{{thread_m_blocks}}, "
+            "{{thread_n_blocks}}, "
+            "{{thread_k_blocks}}, "
+            "{{'true' if m_block_size_8 else 'false'}}, "
+            "{{stages}}, "
+            "{{'true' if has_act_order else 'false'}}, "
+            "{{'true' if has_zp else 'false'}}, "
+            "{{group_blocks}}, "
+            "{{'true' if is_zp_float else 'false'}}>"
+            "( MARLIN_KERNEL_PARAMS );")
+
+# int8 with zero point case (vllm::kU8) is also supported,
+# we don't add it to reduce wheel size.
+SCALAR_TYPES = ["vllm::kU4", "vllm::kU4B8", "vllm::kU8B128"]
+THREAD_CONFIGS = [(128, 128, 256), (64, 256, 256), (64, 128, 128)]
+
+THREAD_M_BLOCKS = [0.5, 1, 2, 3, 4]
+GROUP_BLOCKS = [0, -1, 2, 4, 8]
```

Comment:
What is 0 for vs -1? Would be useful to have comment on how to expand these values to group_size. It seems multiply by 16?

Reply:
1. `group_blocks=0` is the act order case, this is the setting of gptq marlin dense linear kernel. 
2. Yes. `group_blocks=8` means `group_size=128`, this is also the setting of gptq marlin dense linear kernel.